src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -232,41 +281,60 @@ end = struct
       ; current_root= initial_root_transition }
     in
     let transition_graph = Transition_cache.create () in
+    let result = Mvar.create () in
     let%bind synced_db =
       let root_sync_ledger =
         Root_sync_ledger.create ledger_db ~parent_log:t.logger
       in
       sync_ledger t ~root_sync_ledger ~transition_graph ~transition_reader
+        ~result
       |> don't_wait_for ;
       let%map synced_db = Root_sync_ledger.valid_tree root_sync_ledger in
       Root_sync_ledger.destroy root_sync_ledger ;
       synced_db
     in
+    let%bind snarked_ledger_hash, expected_merkle_root, scan_state =
+      Mvar.take result
```

Comment:
Why use an Mvar if you are only taking once? Seems like what you want is just a ref.

Reply:
I am using Mvar because it could be re-set multiple times. I think I would just keep it like that in this PR, since your comment on 'request aux separately' would cause this part of code to be changed anyway. And I don't to block this PR.